features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/SystemRequirementConfiguration.kt

```diff
@@ -0,0 +1,155 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay
+
+import android.os.Handler
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.api.feature.FeatureSdkCore
+import com.datadog.android.core.InternalSdkCore
+import com.datadog.android.core.internal.utils.submitSafe
+import com.datadog.android.sessionreplay.internal.prerequisite.CPURequirementChecker
+import com.datadog.android.sessionreplay.internal.prerequisite.MemoryRequirementChecker
+import com.datadog.android.sessionreplay.internal.prerequisite.SystemRequirementChecker
+
+/**
+ * This class defines the system requirements necessary to enable the Session Replay feature.
+ *
+ * @param minCPUCores The minimum CPU core number.
+ * @param minRAMSizeMb The minimum RAM in megabytes.
+ */
+class SystemRequirementConfiguration internal constructor(
+    internal val minCPUCores: Int,
+    internal val minRAMSizeMb: Int
+) {
+
+    internal fun runIfMeetRequirements(
+        sdkCore: FeatureSdkCore,
+        uiHandler: Handler,
+        internalLogger: InternalLogger,
+        runnable: () -> Unit
+    ) {
+        val checkers = listOf(
+            CPURequirementChecker(minCPUCores, internalLogger = internalLogger),
+            MemoryRequirementChecker(minRAMSizeMb, internalLogger = internalLogger)
+        )
+        (sdkCore as? InternalSdkCore)?.getPersistenceExecutorService()
+            ?.submitSafe(OPERATION_NAME, internalLogger) {
+                if (checkAllMinimumRequirements(checkers)) {
+                    uiHandler.post {
+                        runnable()
+                    }
+                } else {
+                    internalLogger.log(
+                        level = InternalLogger.Level.INFO,
+                        listOf(InternalLogger.Target.TELEMETRY, InternalLogger.Target.USER),
+                        messageBuilder = {
+                            "Session replay is disabled because the system doesn't meet the minimum " +
+                                "Session Replay requirements"
```

Comment:
Yes, this log actually already targets to telemetry, and the concrete requirements are already logged in the `addtionalProperties`, please check ` getCheckerReport(checkers)` function.

Reply:
But is it useful to have it in the telemetry? We may have a lot of entries potentially.

Reply:
I am not sure we can have an action upon this telemetry, but maybe it can give us a visibility how frequently this system requirements work and what's the devices are mostly filtered out?